### ISSUE TITLE
Boot2Docker windows failing to ssh into the guest machine

### ISSUE THREAD

**Issue by lucianenache** [boot2docker/boot2docker#767]
##### I following these steps:

`boot2docker init`
`boot2docker up`

##### In virtualbox I see the machine running in the preview window, even tough at a certain point in the process I get the following error like ~30 times: (run in -v mode)

```bash
o2015/03/01 16:01:57 executing: C:\Program Files (x86)\Git\bin\ssh.exe ssh -o IdentitiesOnly=yes -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -p 2022 -i c:\Program Files (x86)\Vim\.ssh\id_boot2docker docker
@localhost grep tcp:// /proc/$(cat /var/run/docker.pid)/cmdline
Error requesting socket: exit status 255
```

##### Then at the end the program exits with the following message:

```bash
Started.
  Trying to get Docker socket one more time
2015/03/01 16:02:01 executing: C:\Program Files (x86)\Git\bin\ssh.exe ssh -o IdentitiesOnly=yes -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -p 2022 -i c:\Program Files (x86)\Vim\.ssh\id_boot2docker docker@
localhost grep tcp:// /proc/$(cat /var/run/docker.pid)/cmdline
Error requesting socket: exit status 255
2015/03/01 16:02:02 executing: C:\Program Files (x86)\Git\bin\ssh.exe ssh -o IdentitiesOnly=yes -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -p 2022 -i c:\Program Files (x86)\Vim\.ssh\id_boot2docker docker@
localhost tar c /home/docker/.docker/*.pem
Docker client does not run on Windows for now. Please use
    "c:\Program Files\Boot2Docker for Windows\boot2docker.exe" ssh
to SSH into the VM instead.
```

##### Any help would be greatly appreciated.

**Comment by SvenDowideit** [comment]
I would start the vm again using the virtualbox GUI - that way you can see if the VM is hanging somewhere. 

The other thing worth trying sometimes is to delete them and start again - sometimes it just gets confused.

What version of windows are you running?

**Comment by lucianenache** [comment]
So I've already tried both solutions before posting, I am currently running on windows 7 - 64bit , the funny thing is that if I try to run the image from the virtual box it runs flawlessly, and also when boot2docker fires up the image, I can see inside virtualbox trough the preview window that the image loads and it ends up in a waiting state with the shell open

**Comment by SvenDowideit** [comment]
I wonder if something went wrong in the creation of the NAT based port forwarding. 

If you use the vbox GUI, you should be able to check if there is a mapping from the vm's port 22 and the host's 2022 :/

**Comment by lucianenache** [comment]
So the port forwarding is there, on top of that, why does the image have two network adapters configured?

the first one is the NAT with port forwarding for the SSH protocol (it misses the guest ip is that ok?)
the second adapter is a host only adapter.

On top of that when I opened the image settings there were two warnings (1st because the image was using all 8 cores that my machine has, it disappeared when I allocated only 4, and the second one because the video memory was too low : 1mb also disappeared when I've increased it)

Beside that I was thinking, I had vim/git console and ssh already installed before installing boot2docker, maybe it conflicts with those tools.
I was thinking to remove everything and do a fresh reinstall of only boot2docker, I'll let you know how it went)

**Comment by SvenDowideit** [comment]
yes, on windows, there's a strong assumption that you have exactly the right versions of MSYS-git installed, so that might be the problem

**Comment by lucianenache** [comment]
That solved it :smile: so for all future reference, if you had already **git bash** and/or **ssh** and/or **virtualbox**, it is advised to do a clean install by removing the named tools and whatever configu/resilient files or folders might have created in program files or your user folder.

Cheers

**Comment by kevinhughes27** [comment]
I am having the same issue on a fresh install (win 7 x64 professional). Looking at the vm directly using the virtual box gui shows that it is hanging on `Need TLS certs for boot2docker`
